src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -220,6 +221,14 @@ private void callBlockHandler(BlockBreakEvent e, ItemStack item, List<ItemStack>
             }
 
             drops.addAll(sfItem.getDrops());
+            // Partial fix for #4087 - We don't want the inventory to be usable post break, close it for anyone still inside
+            // The main fix is in SlimefunItemInteractListener preventing opening to begin with
+            // Close the inventory for all viewers of this block
+            // TODO(future): Remove this check when MockBukkit supports viewers
+            if (!Slimefun.instance().isUnitTest()) {
+                BlockStorage.getInventory(e.getBlock()).toInventory().getViewers().forEach(HumanEntity::closeInventory);
+            }
+            // Remove the block data
             BlockStorage.clearBlockInfo(e.getBlock());
```

Comment:
I will say this scares me 😅 sense this just kinda goes around the existing one tick delay that we don't know the reason behind existing

Reply:
huh? This is existing code, this is how we remove the data on break (which triggers the cleanup a tick later). This is normal?

Reply:
Ah I misunderstood, mb, review on my phone at like midnight 💀